### DISCUSSION TITLE
Tuya reworking Cloud API | API Adjustment Announcement

### DISCUSSION THREAD

**Discussion by uzlonewolf** [jasonacox/tinytuya#371] [General]
> On June 20, 2023, Tuya has upgraded IoT Core to provide more general and basic APIs at the device layer. Customers in various industries can flexibly integrate with Tuya's device capabilities. For more information, see [IoT Core Service](https://us.iot.tuya.com/cloud/products/detail?abilityId=1442730014117204014). Moreover, the platform has also adjusted some API directories. For more information, see [Adjustment of API Directories](https://developer.tuya.com/en/docs/cloud/device-timer?id=Kcok02xpc6xp5).
If you want to view the legacy API documentation, go to [Legacy Cloud Service APIs (Archived)](https://developer.tuya.com/en/docs/legacy-reference-of-cloud-service-apis).

I have no idea what this is going to mean for permissions.  [Adjustment of API Directories](https://developer.tuya.com/en/docs/cloud/device-timer?id=Kcok02xpc6xp5) says:
> Tuya will synchronize the subscription relationship of existing customers before and after API adjustment. Existing customers’ permission to call APIs remains unaffected.

But I'm not sure what's going to happen for new users.

**Comment by jasonacox** [comment]
Interesting. It's definitely not clear from their communication what will actually happen.
